src/EditorFeatures/Core/Venus/ContainedDocumentBuffers.cs

```diff
@@ -0,0 +1,1097 @@
+
+            if (change.Span.End <= visibleTextSpan.Start)
+            {
+                return true;
+            }
+
+            if (visibleTextSpan.End <= change.Span.Start)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private static IEnumerable<TextChange> GetSubTextChanges(ITextDifferencingService diffService, SourceText originalText, TextChange changeInOriginalText, TextSpan visibleSpanInOriginalText)
+        {
+            using var changes = SharedPools.Default<List<TextChange>>().GetPooledObject();
+
+            var leftText = originalText.ToString(changeInOriginalText.Span);
+            var rightText = changeInOriginalText.NewText;
+            var offsetInOriginalText = changeInOriginalText.Span.Start;
+
+            if (TryGetSubTextChanges(originalText, visibleSpanInOriginalText, leftText, rightText, offsetInOriginalText, changes.Object))
+            {
+                return changes.Object.ToList();
+            }
+
+            return GetSubTextChanges(diffService, originalText, visibleSpanInOriginalText, leftText, rightText, offsetInOriginalText);
+        }
+
+        private static bool TryGetSubTextChanges(
+            SourceText originalText, TextSpan visibleSpanInOriginalText, string leftText, string rightText, int offsetInOriginalText, List<TextChange> changes)
+        {
+            // these are expensive. but hopefully we don't hit this as much except the boundary cases.
+            using var leftPool = SharedPools.Default<List<TextSpan>>().GetPooledObject();
+            using var rightPool = SharedPools.Default<List<TextSpan>>().GetPooledObject();
+
+            var spansInLeftText = leftPool.Object;
+            var spansInRightText = rightPool.Object;
+
+            if (TryGetWhitespaceOnlyChanges(leftText, rightText, spansInLeftText, spansInRightText))
+            {
+                for (var i = 0; i < spansInLeftText.Count; i++)
+                {
+                    var spanInLeftText = spansInLeftText[i];
+                    var spanInRightText = spansInRightText[i];
+                    if (spanInLeftText.IsEmpty && spanInRightText.IsEmpty)
+                    {
+                        continue;
+                    }
+
+                    var spanInOriginalText = new TextSpan(offsetInOriginalText + spanInLeftText.Start, spanInLeftText.Length);
+                    if (TryGetSubTextChange(originalText, visibleSpanInOriginalText, rightText, spanInOriginalText, spanInRightText, out var textChange))
+                    {
+                        changes.Add(textChange);
+                    }
+                }
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private static IEnumerable<TextChange> GetSubTextChanges(ITextDifferencingService diffService,
+            SourceText originalText, TextSpan visibleSpanInOriginalText, string leftText, string rightText, int offsetInOriginalText)
```

Comment:
Either keep the parameters all on one line or put them all on their own lines. (There's a refactoring to do this easily.)